Source/WebKit/WebProcess/cocoa/VideoFullscreenManager.mm

```diff
@@ -127,6 +128,12 @@ static FloatRect inlineVideoFrame(HTMLVideoElement& element)
     , m_playbackSessionManager(playbackSessionManager)
 {
     WebProcess::singleton().addMessageReceiver(Messages::VideoFullscreenManager::messageReceiverName(), page.identifier(), *this);
+    auto representationFactory = [page = WeakPtr { page }] (TextTrackRepresentationClient& client) {
+        auto textTrackRepresentation = makeUnique<WebKit::TextTrackRepresentationCocoa>(client);
+        textTrackRepresentation->setPage(page);
+        return textTrackRepresentation;
+    };
+    WebCore::TextTrackRepresentationCocoa::representationFactory() = WTFMove(representationFactory);
```

Comment:
Shouldn't we be checking the `blockMediaLayerRehostingInWebContentProcess()` setting in here somewhere, so that we have an easy way of turning this path off if there are bugs?

Reply:
Also, I think this is just incorrect code; you're setting a global but passing into the lambda a specific WebPage. It might be okay so long as there is never any more than one WebPage per process, but it has the potential to seriously break if that is ever not true.

I would consider having the factory method take a HTMLMediaElement&. From there you can get the document().page(), convert that to a WebPage, and continue normally.

Reply:
> Shouldn't we be checking the `blockMediaLayerRehostingInWebContentProcess()` setting in here somewhere, so that we have an easy way of turning this path off if there are bugs?

Fixed in latest patch.

Reply:
> Also, I think this is just incorrect code; you're setting a global but passing into the lambda a specific WebPage. It might be okay so long as there is never any more than one WebPage per process, but it has the potential to seriously break if that is ever not true.
> 
> I would consider having the factory method take a HTMLMediaElement&. From there you can get the document().page(), convert that to a WebPage, and continue normally.

That is a good catch! I have addressed this in the latest patch.